### DISCUSSION TITLE
Is it possible to change the panels configurations on the fly?

### DISCUSSION THREAD

**Discussion by witnessmenow** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#158] [Q&A]
I'm just wondering is it possible for a sketch to change its panel configurations on the fly, or would the board need to be restarted in between?

I'm looking at making some of my examples available using the web flasher (Visit website, plug in your ESP32, click flash, no installation of Arduino or even libraries needed) and I'm just wondering what options are available for the config part.

I made a basic version on a gist to test it and it doesn't work past the first one, display goes blank, doesn't even reach line 121.

https://gist.github.com/witnessmenow/f05c3326415325a22f436f00f6dab0cb 

Thanks!

**Comment by mrcodetastic** [comment]
At this stage not possible for all configuration changes, especially for things like clkphase which are defined on DMA setup once only. Such dynamic changes aren't really required - but I see your point in iterating through all configuration to find the one that works?

Cleanest solution may be to simply iterate through config after a esp.reboot() or something.

Use EEPROM to store the last configuration state tested.

**Reply by witnessmenow** [reply]
Cheers, just to confirm the limitation boils down to that you can only init the library once? not a problem at all just want to know what I'm working with!

Other than these two configs, is there anything else you think could be toggled? These are the two that impact my panels here.

**Comment by mrcodetastic** [comment]
Yep. The library allocates the DMA hardware and memory etc. on setup once. Once up and running / allocated there's no going back.

Other than brightness and latch blanking, everything else is set on boot once via. the 'mxconfig' HUB75_I2S_CFG struct.

**Reply by mrcodetastic** [reply]
Looks like @vortigont has implemented the ability to change without reboot #165
